azurerm/config.go

```diff
@@ -323,11 +323,19 @@ func (c *ArmClient) configureClient(client *autorest.Client, auth autorest.Autho
 	setUserAgent(client)
 	client.Authorizer = auth
 	//client.RequestInspector = azure.WithClientID(clientRequestID())
-	client.Sender = autorest.CreateSender(withRequestLogging())
+	client.Sender = buildSender()
 	client.SkipResourceProviderRegistration = c.skipProviderRegistration
 	client.PollingDuration = 60 * time.Minute
 }
 
+func buildSender() autorest.Sender {
```

Comment:
Nitpick: I think that keeping the interface closer to `autorest.CreateSender` (where it accepts existing sender as argument) would help in terms of clarity/readability and explicitness.